modules/mixcr/1.2/mixcr.smk

```diff
@@ -0,0 +1,320 @@
+        input:
+            mixcr_finished = rules._mixcr_run.output.txt,
+            mixcr_chains = rules._mixcr_run.output.results,
+            mixcr_results = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.txt",
+            script = CFG["igblast_scripts"]["mixcr2fasta"]
+        output:
+            fasta = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.VDJseq.fasta",
+            seq_info = temp(CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.regions.txt")
+        shell:
+            "{input.script} -i {input.mixcr_results} -o {output.fasta} -s {output.seq_info}"
+
+    receptor_dict = {"IGH": "ig", "IGK": "ig", "IGL": "ig", "TRA": "tr", "TRB": "tr", "TRD":"tr", "TRG": "tr"}
+    run_dict = {"IGH":"Ig", "IGK":"Ig", "IGL":"Ig", "TRA":"TCR","TRB":"TCR","TRD":"TCR","TRG":"TCR"}
+
+    rule _igblastn_run:
+        input:
+            fasta = rules._mixcr_to_fasta.output.fasta,
+            ig_db = reference_files("genomes/no_build/igblast/database/imgt_database.success")
+        output:
+            db = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.igblastn.fmt7"
+        params:
+            receptor_type = lambda wildcards: run_dict[wildcards.chain],
+            aux = reference_files("downloads/igblast/optional_file/human_gl.aux"),
+            gdv = lambda wildcards: (reference_files("genomes/no_build/igblast/database/imgt_human_" + receptor_dict[wildcards.chain] + "_v.ndb")).replace(".ndb",""),
+            gdj = lambda wildcards: (reference_files("genomes/no_build/igblast/database/imgt_human_" + receptor_dict[wildcards.chain] + "_j.ndb")).replace(".ndb",""),
+            gdd = lambda wildcards: (reference_files("genomes/no_build/igblast/database/imgt_human_" + receptor_dict[wildcards.chain] + "_d.ndb")).replace(".ndb",""),
+            run_flags = CFG["options"]["igblast_run"]["run_flags"],
+            form = "7 std btop " + CFG["options"]["igblast_run"]["form"]
+        conda:
+            CFG["conda_envs"]["igblast"]
+        shell:
+            op.as_one_line("""
+            igblastn -query {input.fasta} -out {output.db} 
+            -ig_seqtype {params.receptor_type} -organism human 
+            -auxiliary_data {params.aux} 
+            -germline_db_V {params.gdv} -germline_db_J {params.gdj} -germline_db_D {params.gdd} 
+            {params.run_flags} -outfmt '{params.form}' -domain_system imgt
+            """)
+
+    rule _update_mixcr_results:
+        input:
+            db = rules._igblastn_run.output.db,
+            og_mixcr = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.txt",
+            seq_info = rules._mixcr_to_fasta.output.seq_info,
+            script = CFG["igblast_scripts"]["igblastn2mixcr"]
+        output:
+            txt = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.status.txt"
```

Comment:
I'd like to suggest that `txt = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.igblast.txt"` would be a more informative name for this output file.

Reply:
Updated output file to `txt = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.igblast.txt"`